src/forwarder/Forwarder.sol

```diff
@@ -292,34 +298,15 @@ contract Forwarder is IForwarder, Ownable, Nonces, Multicall, SelfPermit, Reentr
         assert(abi.decode(result, (uint256)) == orderId);
     }
 
-    /**
-     * @dev Validates the function selector of the encoded function call data.
-     * @param data The encoded function call data.
-     * @param functionSelector The expected function selector.
-     */
-    function _validateFunctionSelector(bytes calldata data, bytes4 functionSelector) internal pure {
-        bytes4 selector = bytes4(data[:4]);
-        if (selector != functionSelector) revert UnsupportedCall();
-    }
-
-    /**
-     * @dev Validates the forward request by checking the oracle price, target address, deadline,
-     *      verifying the price attestation, checking the nonce, and ensuring that the signer of the request is valid.
-     * @param metaTx The meta transaction containing the user address, target contract, encoded function call data,
-     *      deadline, nonce, payment token oracle price, and the signature components (v, r, s).
-     */
-    function _validateForwardRequest(ForwardRequest calldata metaTx) internal {
+    function _validateForwardRequest(OrderForwardRequest calldata metaTx) internal {
```

Comment:
Take signature parameters individually, including hash, so both order and cancel requests can validate signatures